tests/test_config.py

```diff
@@ -1275,3 +1275,15 @@ def test_environment(tmp_path: Path) -> None:
 
         assert sub.environment["PassThisEnv"] == "abc"
         assert "TestValue2" not in sub.environment
+
+
+def test_mkosi_version_run(tmp_path: Path) -> None:
+    d = tmp_path
+
+    version = d / "mkosi.version"
+    version.write_text("#!/bin/sh\necho '1.2.3'\n")
+    version.chmod(0o755)
```

Comment:
Can you add another assert where this file is made non-executable to check that the error is raised correctly?